src/core/util_buffers.h

```diff
@@ -203,38 +204,39 @@ template<typename T>
 class CircularBuffer
 {
 public:
-
     CircularBuffer(size_t numElements) :
-    m_read(0),
-    m_write(0),
-    m_numElements(numElements+1) // need 1 element to pad
+        m_read(0),
+        m_write(0),
+        m_numElements(numElements+1) // need 1 element to pad
     {
+        // allocate
         m_elements = new T[m_numElements];
     }
 
     ~CircularBuffer()
     {
-        if(m_elements != NULL)
-        {
-            delete[] m_elements;
-            m_elements = NULL;
-        }
+        SAFE_DELETE_ARRAY( m_elements );
+        // if(m_elements != NULL)
+        // {
+        //    delete[] m_elements;
+        //    m_elements = NULL;
+        // }
     }
 
     // put one element
     // returns number of elements successfully put
     size_t put(T element)
     {
-        if((m_write + 1)%m_numElements == m_read)
+        // check for overflow
+        if( (m_write + 1)%m_numElements == m_read )
         {
-            // no space
+            // no space | 1.5.0.1 (ge) added log message
+            EM_log( CK_LOG_WARNING, "[circular-buffer]: buffer full, dropping items!" );
```

Comment:
might be nicer to report the error back up to the system that uses it and allow that to print the error as needed -- as-is, even if you saw this error you wouldnt necessarily know it was associated with WvOut or something else

Reply:
agree; this is a stopgap; outside warning reporting would certainly be more informative. have opened issue #327